modules/component-store/spec/component-store.spec.ts

```diff
@@ -511,6 +541,45 @@ describe('Component Store', () => {
     );
   });
 
+  describe('patches the state', () => {
+    interface State {
+      value1: string;
+      value2: { foo: string };
+    }
+    const INIT_STATE: State = { value1: 'value1', value2: { foo: 'bar' } };
+    let componentStore: ComponentStore<State>;
+
+    beforeEach(() => {
+      componentStore = new ComponentStore(INIT_STATE);
+    });
+
+    it(
+      'with a specific value',
+      marbles((m) => {
+        componentStore.patchState({ value1: 'val1' });
+
+        m.expect(componentStore.state$).toBeObservable(
+          m.hot('s', {
+            s: { ...INIT_STATE, value1: 'val1' },
+          })
+        );
+      })
+    );
+
+    it(
+      'with a value based on the previous state',
+      marbles((m) => {
+        componentStore.patchState(() => ({ value2: { foo: 'fooBar' } }));
```

Comment:
Should this test use the current state to build a new state?

e.g.

```suggestion
        componentStore.patchState((state) => ({ value2: { foo: state.value2.foo + "2" } }));
```

Reply:
sure 🙂